torch_xla/experimental/eager.py

```diff
@@ -1,9 +1,48 @@
+import functools
+
 import torch_xla
 
 
 def eager_mode(enable: bool):
   """Configure torch_xla's default executation mode.
+
   Under eager mode only functions that was `torch_xla.compile`d will be
   traced and compiled. Other torch ops will be executed eagerly.
   """
   torch_xla._XLAC._set_use_eager_mode(enable)
+
+
+def is_eager_mode() -> bool:
+  """Return True if torch_xla is currently under eager mode
+  """
+  return torch_xla._XLAC._get_use_eager_mode()
+
+
+def compile(func):
+  """Compile the func with Lazy Tensor.
+
+  Return the optimized function that takes exact same input. Compile will
+  run the target func under the tracing mode using Lazy tensor.
+  """
+
+  @functools.wraps(func)  # Keep function's name, docstring, etc.
+  def wrapper(*args, **kwargs):
```

Comment:
so is the mechanism for caching the graph is already there right so no need to do anything extra?

Reply:
we still need to trace the whole model(run all python code), we just skip XLA compilation and lowering to HLO part.

This `compile` does not modify the function bytecode or transform the function in anyway besides enabling the tracing mode. It is actually more accurate to call it `trace` but `compile` is more align with pytorch API.